framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -63,4 +93,98 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 			await this._sidechainAPIClient.disconnect();
 		}
 	}
+	public async getCertificateFromAggregateCommit(
+		aggregateCommit: AggregateCommit,
+	): Promise<Certificate> {
+		const block = await this._sidechainAPIClient.invoke<chain.Block>('chain_getBlockByHeight', {
```

Comment:
You can also get the blockHeader from the stored blockHeaders and get by height as we are storing it here